qiskit_nature/second_q/hamiltonians/lattices/kagome_lattice.py

```diff
@@ -0,0 +1,417 @@
+        """Return a list consisting of the edges in the bulk, which don't cross the boundaries.
+
+        Returns:
+            list[tuple[int, int, complex]] : list of weighted edges that don't cross the boundaries.
+        """
+        size = self._size
+        edge_parameter = self._edge_parameter
+        num_sites_per_cell = self._num_sites_per_cell
+        list_of_edges = []
+        rows, cols = size
+        unit_cell_coordinates = list(product(*map(range, size)))
+
+        for x, y in unit_cell_coordinates:
+            # each cell is indexed by its leftmost lattice site
+            cell_a_idx = self._coordinate_to_index(np.array([x, y]))
+
+            # indices of sites within unit cell
+            cell_a_0 = num_sites_per_cell * cell_a_idx
+            cell_a_1 = num_sites_per_cell * cell_a_idx + 1
+            cell_a_2 = num_sites_per_cell * cell_a_idx + 2
+
+            # connect sites within a unit cell
+            list_of_edges.append((cell_a_0, cell_a_1, edge_parameter))
+            list_of_edges.append((cell_a_1, cell_a_2, edge_parameter))
+            list_of_edges.append((cell_a_2, cell_a_0, edge_parameter))
+
+            # one cell east if not at the east boundary
+            if x != rows - 1:
+                cell_b_idx = self._coordinate_to_index(np.array([x, y]) + np.array([1, 0]))
+                cell_b_0 = num_sites_per_cell * cell_b_idx
+                list_of_edges.append((cell_a_1, cell_b_0, edge_parameter))
+
+            # one cell north if not at the north boundary
+            if y != cols - 1:
+                cell_b_idx = self._coordinate_to_index(np.array([x, y]) + np.array([0, 1]))
+                cell_b_0 = num_sites_per_cell * cell_b_idx
+                list_of_edges.append((cell_a_2, cell_b_0, edge_parameter))
+
+            # one cell west and north if not at west north boundary
+            if x != 0 and y != cols - 1:
+                cell_b_idx = self._coordinate_to_index(np.array([x, y]) + np.array([-1, 1]))
+                cell_b_1 = num_sites_per_cell * cell_b_idx + 1
+                list_of_edges.append((cell_a_2, cell_b_1, edge_parameter))
+
+        return list_of_edges
+
+    def _boundary_edges(self) -> list[tuple[int, int, complex]]:
+        """Return a list consisting of the edges that cross the boundaries
+            depending on the boundary conditions.
+
+        Raises:
+            ValueError: Given boundary condition is invalid values.
+
+        Returns:
+            list[tuple[int, int, complex]]: list of weighted edges that cross the boundaries.
+        """
+        list_of_edges = []
+        size = self._size
+        edge_parameter = self._edge_parameter
+        num_sites_per_cell = self._num_sites_per_cell
+        boundary_condition = self._boundary_condition
+        rows, cols = size
+        # add edges when the boundary condition is periodic.
+        if boundary_condition == BoundaryCondition.PERIODIC:
```

Comment:
In the case of the `SquareLattice` we support either a single `BoundaryCondition` value (for none or full periodicity) or a pair to enable only-x or only-y perioidicity.
Is this something you consider adding or is this out-of-scope for you at this time? The latter is fine, I just wanted to ask :+1:

Reply:
For the former option, how should we handle the boundary that go diagonally (as in the last if-block inside `_bulk_edges`) when the boundary condition value enables either x or y periodicity only?

Reply:
When you have only x or y, there should not be any diagonal periodicity. Think of it like a band going either along the x or y direction.
Once you have periodicity in both, that's where the "diagonal" case arises (because instead you would only get a "plus-shape" of the the x- and y-band overlapping).

So in short:
- periodic only in x: only horizontal periodicity
- periodic only in y: only vertical periodicity
- periodic in both: both + diagonal